[Issue 5562][docs]Modify schema compatibility check doc (#5757)
apache/pulsar: site2/docs/schema-understand.md

3. The broker looks up the schema in the schema storage to check if it is already a registered schema. 
   
4. If yes, the broker skips the schema validation since it is a known schema, and returns the schema version to the producer.

5. If no, the broker judge whether this namespace can automatically create schema.
   isAllowAutoUpdateSchema can be modified by Admin-Api and REST-API 
  

> 6. If yes, the broker validates the schema based on the schema compatibility check strategy defined for the topic. 

Author: yes, we can delete the line 546 👍 :)

3. The broker looks up the schema in the schema storage to check if it is already a registered schema. 
   
4. If yes, the broker skips the schema validation since it is a known schema, and returns the schema version to the producer.

5. If no, the broker verifies whether a schema can be automatically created in this namespace:

  * If `isAllowAutoUpdateSchema` sets to **true**, then a schema can be created, and the broker validates the schema based on the schema compatibility check strategy defined for the topic.
  
  * If `isAllowAutoUpdateSchema` sets to **false**, then a schema can not be created, and the consumer is rejected to connect to the broker.
  
**Tip**:

`isAllowAutoUpdateSchema` can be set via **Pulsar admin API** or **REST API.** 

> For how to set `isAllowAutoUpdateSchema` via Pulsar admin API, see [Manage AutoUpdate Strategy](schema-manage.md/#manage-autoupdate-strategy). 

Reviewer: Seems a little weird between #5 and #6? Do we need to add some contents like "If a schema is updated automatically:” after #5?

2. The consumer connects to the broker with the `SchemaInfo` extracted from the passed-in schema instance.
   

> 3. If the schema is compatible, the consumer will be connected. 

Reviewer: ```suggestion
3. If the schema is compatible, the consumer is connected to the broker. 
```

> 4. If the schema is incompatible, the consumer will be disconnected.

Reviewer: ```suggestion
4. If the schema is incompatible, the consumer is disconnected to the broker.
```
